### PR TITLE
fix: Workaround low level network errors by reconnection to the node

### PR DESCRIPTION
**Pull request by gshep**
@gear-tech/dev

### CODE REVIEW

**Review by mertwole — COMMENTED** [review]
Please rebase onto `main`

File: relayer/src/message_relayer/common/gear/block_listener.rs [hunk 1]
```diff
@@ -46,12 +47,10 @@ impl BlockListener {
     pub fn run<const RECEIVER_COUNT: usize>(self) -> [Receiver<GearBlockNumber>; RECEIVER_COUNT] {
         let (senders, receivers): (Vec<_>, Vec<_>) = (0..RECEIVER_COUNT).map(|_| channel()).unzip();
 
-        tokio::spawn(async move {
-            loop {
-                let res = self.run_inner(&senders).await;
-                if let Err(err) = res {
-                    log::error!("Gear block listener failed: {}", err);
-                }
+        tokio::task::spawn_blocking(move || loop {
```
Comment (mertwole) [inline relayer/src/message_relayer/common/gear/block_listener.rs:50]: Why `spawn_blocking`?

**Review by mertwole — APPROVED** [review]

**Review by mertwole — COMMENTED** [review]
Also PR name is outdated now

File: relayer/src/message_relayer/common/gear/block_listener.rs (hunk 1 above)
Comment (gshep) [reply relayer/src/message_relayer/common/gear/block_listener.rs:50]: Like it is in other places: CheckpointsExtractor and MessageSender

**Pull request merged**